src/main/kotlin/org/dashevo/dpp/statetransition/StateTransitionFactory.kt

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2018-present, Dash Core Team
+ *
+ * This source code is licensed under the MIT license found in the
+ * COPYING file in the root directory of this source tree.
+ */
+
+package org.dashevo.dpp.statetransition
+
+import org.dashevo.dpp.contract.ContractFactory
+import org.dashevo.dpp.contract.ContractStateTransition
+import org.dashevo.dpp.document.Document
+import org.dashevo.dpp.document.DocumentsStateTransition
+import org.dashevo.dpp.errors.InvalidStateTransitionTypeError
+import org.dashevo.dpp.identity.IdentityCreateTransition
+import org.dashevo.dpp.util.Cbor
+
+class StateTransitionFactory() {
+
+    fun createStateTransition(rawStateTransition: MutableMap<String, Any?>, options: Options = Options()): StateTransition {
+        var stateTransition : StateTransition
+        when (StateTransition.Types.getByCode(rawStateTransition["type"] as Int)) {
+            StateTransition.Types.DATA_CONTRACT -> {
+                val dataContract = ContractFactory().createDataContract(rawStateTransition["dataContract"] as MutableMap<String, Any?>)
+
+                stateTransition = ContractStateTransition(dataContract);
+            }
+            StateTransition.Types.DOCUMENTS -> {
+                var index: Int = 0
+                val actions = rawStateTransition["actions"] as List<Any>
+                val documents = (rawStateTransition["documents"] as List<Any>).map {
```

Comment:
Here you can use `.mapIndexed { index, any ->` and remove the `var index`.

Reply:
Thanks, this works.